src/test/java/junmin/netflix/costreportservice/controller/CostReportControllerTest.java

```diff
@@ -0,0 +1,211 @@
+package junmin.netflix.costreportservice.controller;
+
+import com.google.gson.Gson;
+import junmin.netflix.costreportservice.exception.ErrorEnum;
+import junmin.netflix.costreportservice.exception.InvalidProductionException;
+import junmin.netflix.costreportservice.exception.InvalidReportException;
+import junmin.netflix.costreportservice.pojo.EPCostRecord;
+import junmin.netflix.costreportservice.service.EPCostReportService;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.RequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+@WebMvcTest(CostReportController.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class CostReportControllerTest {
+	@Autowired
+	MockMvc mvc;
+	@MockBean
+	EPCostReportService mockService;
+	Gson gson;
+
+	@BeforeAll
+	public void setup(){
+		gson = new Gson();
```

Comment:
the service is the default Spring object mapper. How do you guarantee when spring library or gson library is updated, this test is still passing.

Reply:
1. I used gson is due to its simplicity. But you are right, I do not really need to introduce gson.
2. I think in our POM file, it's good to specify the version for each dependency jar. So any jar upgrade will be explicit upgrade and we could run the whole test before releasing it with new upgraded jars.